pkg/vectorize/date_add/date_add.go

```diff
@@ -13,28 +13,55 @@
 // limitations under the License.
 package date_add
 
-import "github.com/matrixorigin/matrixone/pkg/container/types"
+import (
+	"github.com/matrixorigin/matrixone/pkg/container/nulls"
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+)
 
 var (
-	DateAdd     func([]types.Date, []int64, []int64, []types.Date) []types.Date
-	DatetimeAdd func([]types.Datetime, []int64, []int64, []types.Datetime) []types.Datetime
+	DateAdd       func([]types.Date, []int64, []int64, []types.Date) []types.Date
+	DatetimeAdd   func([]types.Datetime, []int64, []int64, []types.Datetime) []types.Datetime
+	DateStringAdd func(*types.Bytes, []int64, []int64, *nulls.Nulls, *types.Bytes) *types.Bytes
 )
 
 func init() {
 	DateAdd = dateAdd
 	DatetimeAdd = datetimeAdd
+	DateStringAdd = dateStringAdd
 }
 
 func dateAdd(xs []types.Date, ys []int64, zs []int64, rs []types.Date) []types.Date {
 	for i, d := range xs {
-		rs[i] = d.ToTime().AddInterval(ys[i], types.IntervalType(zs[i])).ToDate()
+		rs[i] = d.ToTime().AddInterval(ys[0], types.IntervalType(zs[0])).ToDate()
 	}
 	return rs
 }
 
 func datetimeAdd(xs []types.Datetime, ys []int64, zs []int64, rs []types.Datetime) []types.Datetime {
 	for i, d := range xs {
-		rs[i] = d.AddInterval(ys[i], types.IntervalType(zs[i]))
+		rs[i] = d.AddInterval(ys[0], types.IntervalType(zs[0]))
+	}
+	return rs
+}
+
+func dateStringAdd(xs *types.Bytes, ys []int64, zs []int64, ns *nulls.Nulls, rs *types.Bytes) *types.Bytes {
```

Comment:
so you don't use the varchar-> date cast function because you'd rather return NULL than throw a error, right?

Reply:
in mysql, when the varchar is like "xxxx", not a date/datetime, the function will return null.